PSReadLine/UndoRedo.cs

```diff
@@ -75,9 +75,10 @@ public static void Undo(ConsoleKeyInfo? key = null, object arg = null)
                 }
                 _singleton._edits[_singleton._undoEditIndex - 1].Undo();
                 _singleton._undoEditIndex--;
+
                 if (_singleton._options.EditMode == EditMode.Vi && _singleton._current >= _singleton._buffer.Length)
                 {
-                    _singleton._current = Math.Max(0, _singleton._buffer.Length - 1);
+                    _singleton._current = Math.Max(0, _singleton._buffer.Length + ViEndOfLineFactor);
```

Comment:
Is this change related? Is it tested?

Reply:
It's related. It's tested by [line 318](https://github.com/PowerShell/PSReadLine/pull/1528/files#diff-588be69a09431df5cd53b7808a683dc5R318) and [line 324](https://github.com/PowerShell/PSReadLine/pull/1528/files#diff-588be69a09431df5cd53b7808a683dc5R324) for insert mode and command mode respectively.